build/gulp.shared.js

```diff
@@ -1,12 +1,24 @@
+const gulp = require('gulp')
 const exec = require('child_process').exec
+const rimraf = require('gulp-rimraf')
+
+const build = () => {
+  gulp.task('build:shared', gulp.series([buildShared, clean]))
+
+  return gulp.series(['build:shared'])
+}
+
+const clean = () => {
+  return gulp.src('./out/bp/ui-shared/dist', { allowEmpty: true }).pipe(rimraf())
+}
 
 const watch = cb => {
   const shared = exec('yarn && yarn start', { cwd: 'src/bp/ui-shared' }, err => cb(err))
   shared.stdout.pipe(process.stdout)
   shared.stderr.pipe(process.stderr)
 }
 
-const build = cb => {
+const buildShared = cb => {
```

Comment:
I think you could put this in gulp.ui.js directly, since it's the shared UI library

Reply:
I'm worried it might be confusing as both admin and studio can be run in parallel while this one must run before modules

Reply:
Is it possible to avoid linking it directly to modules ? It would make bundle size bigger. You can declare it as an external dependency in the module builder https://github.com/botpress/botpress/blob/master/build/module-builder/src/webpack.ts#L39 and declare typings here: https://github.com/botpress/botpress/blob/master/modules/tsconfig_view.shared.json#L18

We already share some components between the studio and modules, however we do not share them between the studio and the admin, so we have duplicates